src/py/mat3ra/made/tools/build/interface.py

```diff
@@ -1,268 +1,161 @@
-    ) -> List[Interface]:
-        interfaces = self.get_interfaces_for_termination_or_its_index(termination_or_its_index)
-        return array_utils.filter_by_slice_or_index_or_indices(interfaces, slice_or_index_or_indices)
-
-    def remove_duplicate_interfaces(self, strain_mode: StrainModes = StrainModes.mean_abs_strain):
-        for termination in self.terminations:
-            self.remove_duplicate_interfaces_for_termination(termination, strain_mode)
-
-    def remove_duplicate_interfaces_for_termination(
-        self, termination, strain_mode: StrainModes = StrainModes.mean_abs_strain
-    ):
-        def are_interfaces_duplicate(interface1: Interface, interface2: Interface):
-            return interface1.num_sites == interface2.num_sites and np.allclose(
-                interface1.interface_properties[strain_mode], interface2.interface_properties[strain_mode]
-            )
-
-        sorted_interfaces = self.get_interfaces_for_termination_sorted_by_size(termination)
-        filtered_interfaces = [sorted_interfaces[0]] if sorted_interfaces else []
-
-        for interface in sorted_interfaces[1:]:
-            if not any(
-                are_interfaces_duplicate(interface, unique_interface) for unique_interface in filtered_interfaces
-            ):
-                filtered_interfaces.append(interface)
-
-        self.set_interfaces_for_termination(termination, filtered_interfaces)
-
-    def get_interfaces_for_termination_sorted_by_strain(
-        self, termination: Union[int, TerminationType], strain_mode: StrainModes = StrainModes.mean_abs_strain
-    ) -> List[Interface]:
-        return sorted(
-            self.get_interfaces_for_termination(termination),
-            key=lambda x: np.mean(np.abs(x.interface_properties[strain_mode])),
-        )
 
-    def get_interfaces_for_termination_sorted_by_size(
-        self, termination: Union[int, TerminationType]
-    ) -> List[Interface]:
-        return sorted(
-            self.get_interfaces_for_termination(termination),
-            key=lambda x: x.num_sites,
-        )
+class ZSLInterfaceBuilder(BaseInterfaceBuilder):
+    """
+    Creates matching interface between substrate and film using the ZSL algorithm.
+    """
 
-    def get_interfaces_for_termination_sorted_by_strain_and_size(
-        self, termination: Union[int, TerminationType], strain_mode: StrainModes = StrainModes.mean_abs_strain
-    ) -> List[Interface]:
-        return sorted(
-            self.get_interfaces_for_termination_sorted_by_strain(termination, strain_mode),
-            key=lambda x: x.num_sites,
+    def __init__(self, strain_matching_parameters: ZSLStrainMatchingParameters):
+        super().__init__(strain_matching_parameters=strain_matching_parameters)
+        self.strain_matching_parameters = strain_matching_parameters.dict()
+
+    def get_material(self, interface_configuration: InterfaceConfiguration) -> Material:
```

Comment:
We can define BaseStrainMatchingInterfaceBuilder and define get_material, get_sorted_interfaces and get_interface_structures there. Then `ZSLInterfaceBuilder` will inherit from it and only override __init__ and `generator`